src/emqtt_sock.erl

```diff
@@ -87,6 +83,24 @@ send(QuicStream, Data) when is_reference(QuicStream) ->
             Other
     end.
 
+-if(?OTP_RELEASE >= 26).
+send_tcp_data(Sock, Data) ->
+    case gen_tcp:send(Sock, Data) of
+        ok ->
+            self() ! {inet_reply, Sock, ok},
+            ok;
+        Other ->
+            Other
+    end.
```

Comment:
the second clause is really a bug in case reconnect is configured.
hopefully going forward when it's time to ditch OTP25, we can delete these to clauses.